test/PartnerConfiguration.spec.ts

```diff
@@ -343,4 +343,65 @@ describe('Partner Configuration', () => {
     });
   });
+
+  context('Config modification events', () => {
+    it('Should emit the MinDurationChanged event with the correct params', async () => {
+      const NEW_MIN_DURATION = DEFAULT_MIN_DURATION + 1;
+      await expect(PartnerConfiguration.setMinDuration(NEW_MIN_DURATION))
+        .to.emit(PartnerConfiguration, 'MinDurationChanged')
+        .withArgs(DEFAULT_MIN_DURATION, NEW_MIN_DURATION);
+    });
+    it('Should emit the MaxDurationChanged event with the correct params', async () => {
+      const NEW_MAX_DURATION = DEFAULT_MAX_DURATION + 1;
+      await expect(PartnerConfiguration.setMaxDuration(NEW_MAX_DURATION))
+        .to.emit(PartnerConfiguration, 'MaxDurationChanged')
+        .withArgs(DEFAULT_MAX_DURATION, NEW_MAX_DURATION);
+    });
+
+    it('Should emit the UnicodeSupportChanged event with the correct params', async () => {
+      const NEW_IS_UNICODE_SUPPORTED = !DEFAULT_IS_UNICODE_SUPPORTED;
+      await expect(
+        PartnerConfiguration.setUnicodeSupport(NEW_IS_UNICODE_SUPPORTED)
+      )
+        .to.emit(PartnerConfiguration, 'UnicodeSupportChanged')
+        .withArgs(DEFAULT_IS_UNICODE_SUPPORTED, NEW_IS_UNICODE_SUPPORTED);
+    });
+
+    it('Should emit the MinLengthChanged event with the correct params', async () => {
+      const NEW_MIN_LENGTH = DEFAULT_MIN_LENGTH + 1;
+      await expect(PartnerConfiguration.setMinLength(NEW_MIN_LENGTH))
+        .to.emit(PartnerConfiguration, 'MinLengthChanged')
+        .withArgs(DEFAULT_MIN_LENGTH, NEW_MIN_LENGTH);
+    });
+
+    it('Should emit the MaxLengthChanged event with the correct params', async () => {
+      const NEW_MAX_LENGTH = DEFAULT_MAX_LENGTH + 1;
+      await expect(PartnerConfiguration.setMaxLength(NEW_MAX_LENGTH))
+        .to.emit(PartnerConfiguration, 'MaxLengthChanged')
+        .withArgs(DEFAULT_MAX_LENGTH, NEW_MAX_LENGTH);
+    });
+
+    it('Should emit the FeePercentageChanged event with the correct params', async () => {
+      const NEW_FEE_PERCENTAGE = DEFAULT_FEE_PERCENTAGE + 1;
+      await expect(PartnerConfiguration.setFeePercentage(NEW_FEE_PERCENTAGE))
+        .to.emit(PartnerConfiguration, 'FeePercentageChanged')
+        .withArgs(DEFAULT_FEE_PERCENTAGE, NEW_FEE_PERCENTAGE);
+    });
+
+    it('Should emit the DiscountChanged event with the correct params', async () => {
+      const NEW_DISCOUNT = DEFAULT_DISCOUNT + 1;
+      await expect(PartnerConfiguration.setDiscount(NEW_DISCOUNT))
+        .to.emit(PartnerConfiguration, 'DiscountChanged')
+        .withArgs(DEFAULT_DISCOUNT, NEW_DISCOUNT);
+    });
+
+    it('Should emit the MinCommitmentAgeChanged event with the correct params', async () => {
+      const NEW_MIN_COMMITMENT_AGE = DEFAULT_MIN_COMMITMENT_AGE + 1;
+      await expect(
+        PartnerConfiguration.setMinCommitmentAge(NEW_MIN_COMMITMENT_AGE)
+      )
+        .to.emit(PartnerConfiguration, 'MinCommitmentAgeChanged')
+        .withArgs(DEFAULT_MIN_COMMITMENT_AGE, NEW_MIN_COMMITMENT_AGE);
+    });
+  });
```

Comment:
kind of a pain in the ass but... can we move the events strings to a single location?

Reply:
Sure thing